### PR TITLE
Adding back "Datastore" section to docs.

### PR DESCRIPTION
**Pull request by dhermes**

### CODE REVIEW

**Comment by dhermes** [comment]
Was removed in https://github.com/GoogleCloudPlatform/gcloud-python/commit/5cca643db10eee125a378c53437fe14e4f42d48d

[Currently][1] looks like
screen_shot_001
instead of [expected][2]
screen_shot_002

[1]: http://googlecloudplatform.github.io/gcloud-python/master/
[2]: http://googlecloudplatform.github.io/gcloud-python/latest/

File: docs/index.rst [hunk 1]
```diff
@@ -3,6 +3,7 @@
   :hidden:
 
   gcloud-api
+  datastore-api
```
Comment (tseaver) [inline docs/index.rst]: `datastore-api` is misnamed now (it doesn't have any API documentation, and there is no `api` module to which it corresponds).

Comment (dhermes) [inline docs/index.rst]: If you look at what the document corresponds to, it's just a basic description of the package. (See the screenshots in this PR.)

What should I name the file? What should we put in this files?

Comment (tseaver) [inline docs/index.rst]: Maybe `datastore-overview`?  If it is just the information you added back, that seems descriptive enough.

Comment (dhermes) [inline docs/index.rst]: OK cool.

**Comment by dhermes** [comment]
@tseaver I updated the original commit to use `datastore-overview` over `datastore-api`.

**Comment by tseaver** [comment]
LGTM

**Pull request merged**